debian9/BUILD

```diff
@@ -0,0 +1,81 @@
+# Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@base_images_docker//util:run.bzl", "container_run_and_extract")
+load(
+    "@bazel_toolchains//rules/container:docker_toolchains.bzl",
+    "toolchain_container",
+)
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
+load("@io_bazel_rules_docker//container:container.bzl", "container_image")
+load("@io_bazel_rules_docker//contrib:test.bzl", "container_test")
+
+licenses(["notice"])  # Apache 2.0
+
+package(default_visibility = ["//visibility:public"])
+
+toolchain_container(
+    name = "builder",
+    base = "@debian_base//image",
+    files = [":mkimage.sh"],
+    packages = [
+        "debootstrap",
+    ],
+)
+
+container_run_and_extract(
+    name = "rootfs",
+    commands = [
+        " ".join([
+            "/mkimage.sh",
+            "minbase",
+            "stretch",
+        ]),
+    ],
+    docker_run_flags = ["--privileged"],
+    extract_file = "/workspace/rootfs.tar.gz",
+    image = ":builder.tar",
+)
+
+# The overlay directory contains files that should be overlaid into the rootfs at /.
+pkg_tar(
+    name = "overlay",
+    srcs = glob(["overlay/**/*"]),
+    package_dir = "/",
+    strip_prefix = "overlay/",
+)
+
+DEBIAN_ENV = {
+    "DEBIAN_FRONTEND": "noninteractive",
+    "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+    "PORT": "8080",
+}
+
+container_image(
+    name = "image",
+    cmd = "/bin/bash",
+    env = DEBIAN_ENV,
+    tars = [
+        # The overlay tar has to come first to actuall overwrite existing files.
+        ":overlay.tar",
+        # TODO(xingao): Switch to use checked-in file.
```

Comment:
any particular reason why you're not doing this as part of this PR (just curious)?

Reply:
I didn't do this because I want this strategy/modification to the existing way to generate the image to be approved first before I upload any file to the GCS bucket. Since it's approved, I'm gonna do it and modify this PR accordingly.